src/generic/generic-game-factory.vala

```diff
@@ -0,0 +1,54 @@
+// This file is part of GNOME Games. License: GPLv3
+
+public class Games.GenericUriGameFactory : Object, UriGameFactory {
+	private const uint HANDLED_URIS_PER_CYCLE = 5;
+
+	private GameUriAdapter game_for_uri;
```

Comment:
The name would be better if it reflected the fact that it's a GameUriAdapter and not a GameForUri delegate, though it's not _that_ big of a deal.